audiomentations/augmentations/transforms.py

```diff
@@ -1095,16 +1107,30 @@ def apply(self, samples, sample_rate):
 
             clean_rms = calculate_rms(samples[start_sample_index:end_sample_index])
             noise_rms = calculate_rms(noise_samples)
+
             if noise_rms > 0:
-                desired_noise_rms = calculate_desired_noise_rms(
-                    clean_rms, sound_params["snr_in_db"]
-                )
+                if self.noise_power == "relative":
+
+                    desired_noise_rms = calculate_desired_noise_rms(
+                        clean_rms, sound_params["snr_in_db"]
+                    )
+
+                    # Adjust the noise to match the desired noise RMS
+                    noise_samples = noise_samples * (desired_noise_rms / noise_rms)
+
+                    noise_placeholder[start_sample_index:end_sample_index] += noise_samples
+                if self.noise_power == "absolute":
+                    desired_noise_rms_amp = convert_decibels_to_amplitude_ratio(self.absolute_noise_rms_db)
+                    gain = desired_noise_rms_amp / noise_rms
+                    noise_samples = noise_samples * gain
 
-                # Adjust the noise to match the desired noise RMS
-                noise_samples = noise_samples * (desired_noise_rms / noise_rms)
+                    max_noise_samples = np.amax(np.abs(noise_samples))
 
-                noise_placeholder[start_sample_index:end_sample_index] += noise_samples
+                    if max_noise_samples > 1:  # Not optimal, since this will change the rms. Happens primarily to
+                        # noise bursts.
+                        noise_samples = noise_samples / max_noise_samples
```

Comment:
Why do we have this peak normalization here? I think I would personally not add this feature, because as you say it changes the balance between signal and noise so that it doesn't match the snr provided given to \_\_init\_\_.

audiomentations already has several other transforms that can return results with values outside the [-1, 1] range, and I'm not too worried about it, because it can be dealt with at the end of the data augmentation pipeline. Values outside the [-1, 1] range is not really a problem when using float32, but it could result in clipping/distortion when converting it to int16 without normalizing it forst.

Reply:
I added it because some handling noises were repeated quick bursts of sound with silence between the short peaks. This gave an artificially low noise rms and the max value of the peaks after treatment could reach 4,5. I was afraid that the initial speech recording would be hidden because of this. I could maybe calculate the rms of the noises another way, with audio event detection for example

Reply:
I see. Hmm. How about:

1) Divide the short noise into short chunks (maybe 25 ms each? I dunno)
2) Calculate the RMS for each chunk
3) Order the RMS values and take the 90th percentile or something, and treat that as the overall RMS?

Just using my intuition here

Reply:
Maybe it should be configurable with a parameter how we calculate the RMS?

For more or less stationary short noises mean RMS is okay, but for highly nonstationary noises, the more complicated percentile-based approach seems more useful

Reply:
Yes, I was thinking about this approach too :)